src/EnergyPlus/WaterCoils.cc

```diff
@@ -3566,7 +3554,7 @@ void CalcDetailFlatFinCoolingCoil(EnergyPlusData &state,
         //);             } // CoefPointer
         auto const &dry_fin_eff_coef = state.dataWaterCoils->WaterCoil(CoilNum).DryFinEfficncyCoef;
         Real64 DryFinEfficncy_pow = 1.0;
-        for (CoefPointer = 1; CoefPointer <= 5; ++CoefPointer) {
+        for (int CoefPointer = 1; CoefPointer <= 5; ++CoefPointer) {
```

Comment:
What is 5? Is there a symbolic constant that can be used instead?

Reply:
This is number of coefficients of a polynomial function of order 4 generated for water coils in dry operating mode. The size of the array holding the coefficients is defined as `Array1D<Real64> CoefSeries = Array1D<Real64>(5)` [here](https://github.com/NREL/EnergyPlus/blob/develop/src/EnergyPlus/WaterCoils.hh#L581).

Reply:
Replaced `5` with `dry_fin_eff_coef.length()`